engineioxide/src/payload.rs

```diff
@@ -0,0 +1,156 @@
+use std::{io::BufRead, vec};
+
+use crate::{service::ProtocolVersion, errors::Error};
+
+pub const PACKET_SEPARATOR: u8 = b'\x1e';
+
+/// A payload is a series of encoded packets tied together.
+/// How packets are tied together depends on the protocol.
+pub struct Payload<R: BufRead> {
+    reader: R,
+    buffer: Vec<u8>,
+    #[allow(dead_code)]
+    protocol: ProtocolVersion,
+}
+
+type Item = Result<String, Error>;
+
+impl<R: BufRead> Payload<R> {
+    pub fn new(protocol: ProtocolVersion, data: R) -> Self {
+        Payload {
+            reader: data,
+            buffer: vec![],
+            protocol,
+        }
+    }
+
+    #[cfg(feature = "v3")]
+    fn next_v3(&mut self) -> Option<Item> {
+        match self.reader.read_until(b':', &mut self.buffer) {
+            Ok(bytes_read) => {
+                if bytes_read > 0 {
+                    // remove trailing separator
+                    if self.buffer.ends_with(&[b':']) {
+                        self.buffer.pop();
+                    }
+
+                    let buffer = std::mem::take(&mut self.buffer);
+                    let length = match String::from_utf8(buffer) {
+                        Ok(s) => {
+                            if let Ok(l) = s.parse::<usize>() {
```

Comment:
could you just write todo here, this part can be optimized without String allocation. it shouldn't be implemented in the feature, but good to have TODO here for future optimizations

Reply:
So you mean to add a TODO comment here: https://github.com/sleeyax/socketioxide/blob/ft-engineio-v3-support/engineioxide/src/payload.rs#L36-L38

... mentioning that the .parse() to int and string allocation can be optimized?

Reply:
@Totodore mentioned it below. `String::from_utf8` can be replaced with `std::str::from_utf8`

Reply:
I see. I agree, this could be improved upon in a later PR. I'll add a TODO as you suggest.